src/charm.py

```diff
@@ -58,17 +71,26 @@ def _on_update_status(self, _) -> None:
     def _on_config_changed(self, _) -> None:
         """Handle changes in configuration."""
         self.unit.status = ops.MaintenanceStatus("Configuring charm")
+        self._store_password_as_secret()
         self._update_relations()
         self.unit.status = ops.ActiveStatus()
 
+    def _store_password_as_secret(self) -> None:
+        """Store the SMTP password as a secret."""
+        if self._charm_state.password:
+            secret = self.app.add_secret({"password": self._charm_state.password})
```

Comment:
Out of curiosity: did you check if this will create a new secret for every config change?

Reply:
Considering how involved it would be to prevent the secret from changing, I think we can live with this for now. This would require:
- Storing the secret_id in a new peer relation
- Creating the peer relation itself
- Comparing the store value and the new value
What do you think?